src/js/models/layerstructure.js

```diff
@@ -1089,7 +1089,7 @@ define(function (require, exports, module) {
             }, this);
         }.bind(this));
 
-        return this.mergeDeep({
+        return this.merge({
             layers: nextLayers
         });
```

Comment:
In this case, we need `replace` more than `merge`, as nextLayers is a complete set of next layers

```
this.set("layers", nextLayers)
```

If `nextLayers` is a sub-set of `this.layers`, then we should use `merge` on `this.layers`, otherwise `this.layers` will be replaced by an incomplete list of layers.

```
this.set("layers", this.layers.merge(nextLayers))
```